### PR TITLE
AL-1364 Add private tag when baking ami

### PR DESCRIPTION
**Pull request by rruster**
* Add support for the --private option on disco_bake.py
* Add is_private tag to amis

### CODE REVIEW

File: bin/disco_bake.py [hunk 1]
```diff
@@ -121,6 +121,9 @@ def get_parser():
     parser_bake.add_argument('--tag', dest='tags', required=False, action='append', type=str, default=[],
                              help='The key:value pair used to tag the AMI '
                                   '(Example: --tag application:dnext)')
+    parser_bake.add_argument('--private', dest='is_private', action='store_const', const=True,
+                             default=False, help='Tag the baked ami as "private". The AMI will not be visible'
+                                                 ' to disco_deploy.py or disco_aws.py promote.')
```
Comment (aoskotsky-amplify) [inline bin/disco_bake.py]: `disco_aws.py promote` is not a command

**Comment by rruster** [comment]
build

File: tests/unit/test_disco_deploy.py [hunk 2]
```diff
@@ -152,11 +153,12 @@ def mock_group(self, hostclass, min_size=None, max_size=None, desired_size=None,
...
         return group_mock
 
-    def add_ami(self, name, stage, state=u'available'):
+    def add_ami(self, name, stage, state=u'available', is_private=False):
         '''Add one Instance AMI Mock to an AMI list'''
-        ami = self.mock_ami(name, stage, state)
+        ami = self.mock_ami(name, stage, state, is_private)
         assert ami.name == name
-        assert ami.tags.get() == stage
+        assert ami.tags.get('stage') == stage
```
Comment (aoskotsky-amplify) [inline tests/unit/test_disco_deploy.py]: Codacy doesn't like these assert statements

Comment (rruster) [inline tests/unit/test_disco_deploy.py]: Should we remove them?

**Pull request merged**